site/docs/expressions/scalar_functions.md

```diff
@@ -47,12 +46,20 @@ There are four main types of arguments: value arguments, type arguments, require
 | Options  | List of valid string options for this argument               | Required                                                     |
 | Name     | A human-readable name for this argument to help clarify use. | Optional, defaults to a name based on position (e.g. `arg0`) |
 
-#### Optional Enumeration Properties
+## Options
 
-| Property | Description                                                  | Required                                                     |
-| -------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
-| Options  | Priority-ordered list of valid string options for this argument. The pseudo-option will be the default "value" for the enumeration unless a binding specifies a specific value. | Required                                                   |
-| Name     | A human-readable name for this argument to help clarify use. | Optional, defaults to a name based on position (e.g. `arg0`) |
+In addition to arguments each call may specify zero or more options.  These are similar to a required enumeration but more focused on supporting alternative behaviors. Options can be left unspecified and the consumer is free to choose which implementation to use. An example use case might be `OVERFLOW_BEHAVIOR:[OVERFLOW, SATURATE, ERROR]` If unspecified, an engine is free to use any of the three choices or even some alternative behavior (e.g. setting the value to null on overflow). If specified, the engine would be expected to behave as specified or fail. Note, the value of an optional enumeration cannot be used in type derivation.
+
+### Option Precedence
+
+A producer may specify multiple vlaues for an option.  If the producer does so then the consumer must deliver the first behavior in the list of values that the consumer is capable of delivering.  For example, considering overflow as defined above, if a producer specified `[ERROR, SATURATE]` then the consumer must deliver `ERROR` if it is capable of doing so.  If it is not then it may deliver `SATURATE`.  If the consumer cannot deliver either behavior then it is an error and the consumer must reject the plan.
+
+#### Optional Properties
```

Comment:
There's some confusion here since it's not completely clear whether we're talking about the properties of an `algebra.proto::substrait.FunctionOption` message or about the properties of a `simple_extensions_schema.yaml::options` declaration (in general, I think this document would be improved by explicitly listing the substrait message to which these `Property, Description, Required` tables apply).

I would delete `### Optional Precedence``` and hoist this section above the paragraph explaining precedence. That paragraph can then serve as the explanation as to why an `algebra.proto::substrait.FunctionOption` has a list of values instead of a single value.

Reply:
> There's some confusion here since it's not completely clear whether we're talking about the properties of an algebra.proto::substrait.FunctionOption message or about the properties of a simple_extensions_schema.yaml::options declaration (in general, I think this document would be improved by explicitly listing the substrait message to which these Property, Description, Required tables apply).

I had the same confusion.  I think this entire section should probably be moved somewhere else.  This page is describing what belongs in a plan and this section is describing what belongs in extension files.  Those two blocks of documentation should be on separate pages.  The page describing what belongs in a plan can talk about precedence/preference.  The page describing what belongs in the extension yamls can describe the "set of possible values".  I would rather keep this as is and fix this in a future PR.  The proposed change isn't quite right either I think.